src/NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp

```diff
@@ -18,6 +18,12 @@ class er;
 /// \endcond
 
 namespace ylm {
+/// \brief The kind of data points used to construct a `Strahlkorper`
+enum class StrahlkorperConstructorData {
+  RadiusAtCollocationPoints,
+  SpectralCoefficients
```

Comment:
An enum class like this for extending the constructor was requested and I initially chose to name the class and values here very explicitly to make them clear, but I will say that as-is they are verrry long to type (basically have to copy+paste). Suggestions on better (shorter) names are very welcome, e.g. maybe `StrahlkorperData::Radius` and `StrahlkorperData::Coef`?

Reply:
I'm good with the values `Radius`/`Radii` and `SpectralCoefs` but the type is mostly used outside of `Strahlkorper`, so then if I did move it inside, externally this would mean the change `StrahlkorperConstructorDataType::Radius` -> `Strahlkorper::ConstructorDataType::Radius` which is ~ same diff in terms of keystrokes (with a couple extra). Let me know if I'm misunderstanding or if you have a preference, but I will at least shorten the values to `Radius`/`Radii` and `SpectralCoefs`!

Reply:
Ah yeah that's my bad. By "inside" I meant that it's only used within the context/constructor of this class and nowhere else. We actually already have a similar example of an enum used only in a class functions in `src/NumericalAlgorithms/SphericalHarmonics/SpherepackIterator.hpp`, so I'd say move it inside the `Strahlkorper` class. It keeps everything together.

Reply:
See my comment to the Strahlkorper constructor